File: Behavioral/Visitor/Python/Renderer.py
import abc

class AbstractRenderer(metaclass = abc.ABCMeta):
    @abc.abstractmethod
    def renderSphere(self, sphere):
        return
    @abc.abstractmethod
    def renderBox(self, box):
        return

class SceneObject:
    def __init__(self):
        self.name = "Undefined"

    def render(self, renderer):
        print("{0} cannot be rendered with {1}".format(self.name, renderer))

class Sphere(SceneObject):
    def __init__(self, radius):
        self.name = "Sphere"
        self.radius = radius
     
    def render(self, renderer):
        renderer.renderSphere(self)
        
class Box(SceneObject):
    def __init__(self, size):
        self.name = "Box"
        self.size = size
     
    def render(self, renderer):
        renderer.renderBox(self)
        
class ScanlineRenderer(AbstractRenderer):
    def renderSphere(self, sphere):
        print("Rendering a sphere with radius {0} by scanline method.".format(sphere.radius))
        
    def renderBox(self, box):
        print("Rendering a box with size {0} by scanline method.".format(box.size))
        
class RaytracingRenderer(AbstractRenderer):
    def renderSphere(self, sphere):
        print("Rendering a sphere with radius {0} by raytracing method.".format(sphere.radius))
        
    def renderBox(self, box):
        print("Rendering a box with size {0} by raytracing method.".format(box.size))
        
        
sphere = Sphere(3)
box = Box(10)
scanline = ScanlineRenderer()
raytracer = RaytracingRenderer()

sphere.render(scanline)
sphere.render(raytracer)
box.render(scanline)
box.render(raytracer)

File: Behavioral/Visitor/JavaScript/Renderer.js
class Sphere
{
    constructor(radius)
    {
        this.radius = radius;
    }
    
    render(renderer)
    {
        renderer.renderSphere(this);
    }
}

class Box
{
    constructor(size)
    {
        this.size = size;
    }
    
    render(renderer)
    {
        renderer.renderBox(this);
    }
}

class ScanlineRenderer
{
    renderSphere(s)
    {
        console.log("Rendering a sphere with radius ".concat(s.radius.toString()).concat(" by scanline method."));
    }
    
    renderBox(b)
    {
        console.log("Rendering a box with size ".concat(b.size.toString()).concat(" by scanline method."));
    }
}

class RaytracingRenderer
{
    renderSphere(s)
    {
        console.log("Rendering a sphere with radius ".concat(s.radius.toString()).concat(" by raytracing method."));
    }
    
    renderBox(b)
    {
        console.log("Rendering a box with size ".concat(b.size.toString()).concat(" by raytracing method."));
    }
}

// Test

var sphere = new Sphere(3);
var box = new Box(10);
var scanline = new ScanlineRenderer();
var raytracer = new RaytracingRenderer();

sphere.render(scanline);
sphere.render(raytracer);
box.render(scanline);
box.render(raytracer);